src/Storages/MergeTree/MergeTreeData.cpp

```diff
@@ -8605,7 +8605,9 @@ void MergeTreeData::resetSerializationHints(const DataPartsLock & /*lock*/)
         .choose_kind = true,
     };
 
-    const auto & storage_columns = getInMemoryMetadataPtr()->getColumns();
+    const auto meta_data = getInMemoryMetadataPtr();
```

Comment:
What about the other uses of `getInMemoryMetadataPtr()->getColumns()`? In fact the associated issue complains about `updateSerializationHints`

Reply:
Yes, please update the same case a few lines below.

Reply:
Suspicious in the same file: `updateSerializationHints` and `addPartContributionToColumnAndSecondaryIndexSizes` (but this one is not new/recent)